### ISSUE TITLE
All UAPAOT tests not working at all

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#30423]
eg https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Filc~2F/build/20180615.04/workItem/System.Diagnostics.TraceSource.Tests/wilogs

Xunit runs, but produces no exception, and no results xml file. Need to repro this locally.

```
2018-06-15 14:46:28,717: INFO: proc(54): run_and_log_output: Output: C:\dotnetbuild\work\bc4c3087-fcce-4e3c-98df-e4bd1534dbeb\Work\a123ee5f-bdde-42bb-97b2-cdc5a3c54e75\Unzip\native>call xunit.console.netcore.exe System.Diagnostics.TraceSource.Tests.dll  -xml testResults.xml -redirectoutput -notrait category=nonuapaottests -notrait category=nonwindowstests  -notrait category=failing 
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: 
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: C:\dotnetbuild\work\bc4c3087-fcce-4e3c-98df-e4bd1534dbeb\Work\a123ee5f-bdde-42bb-97b2-cdc5a3c54e75\Unzip\native>type Xunit.Console.Output.txt 
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: The system cannot find the file specified.
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: 
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: C:\dotnetbuild\work\bc4c3087-fcce-4e3c-98df-e4bd1534dbeb\Work\a123ee5f-bdde-42bb-97b2-cdc5a3c54e75\Unzip\native>copy /y testResults.xml C:\dotnetbuild\work\bc4c3087-fcce-4e3c-98df-e4bd1534dbeb\Work\a123ee5f-bdde-42bb-97b2-cdc5a3c54e75\Unzip\\ 
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: The system cannot find the file specified.
2018-06-15 14:46:30,329: INFO: proc(54): run_and_log_output: Output: ----- end 14:46:30.33 ----- exit code 1 ----------------------------------------------------------

```

**Comment by danmosemsft** [comment]
@weshaggard

**Comment by weshaggard** [comment]
@joshfree who would be a good person working uap stuff to look into this?

**Comment by joshfree** [comment]
@tommcdon and @MattGal

**Comment by MattGal** [comment]
Note that this issue, while potentially related to #30399, is almost certainly different since for UAPAOT we're building console executables that do not rely on AppX activation.

**Comment by MattGal** [comment]
Spent a little time debugging this on AMD64, I think it IS the exact issue on #30399.  Running the output executable with loader snaps enabled we get:

```
6960:b580 @ 206029390 - LdrpResolveDllName - ENTER: DLL name: C:\Users\mattgal\.dotnet\tools\libc.dll
6960:b580 @ 206029390 - LdrpResolveDllName - RETURN: Status: 0xc0000135
6960:b580 @ 206029390 - LdrpResolveDllName - ENTER: DLL name: C:\Users\mattgal\AppData\Roaming\npm\libc.dll
6960:b580 @ 206029390 - LdrpResolveDllName - RETURN: Status: 0xc0000135
6960:b580 @ 206029390 - LdrpResolveDllName - ENTER: DLL name: C:\Users\mattgal\AppData\Local\Microsoft\WindowsApps\libc.dll
6960:b580 @ 206029390 - LdrpResolveDllName - RETURN: Status: 0xc0000135
6960:b580 @ 206029390 - LdrpResolveDllName - ENTER: DLL name: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\1.0.5\libc.dll
6960:b580 @ 206029390 - LdrpResolveDllName - RETURN: Status: 0xc0000135
6960:b580 @ 206029390 - LdrpSearchPath - RETURN: Status: 0xc0000135
6960:b580 @ 206029390 - LdrpProcessWork - ERROR: Unable to load DLL: "libc.dll", Parent Module: "c:\scratch\uapaot1\bc4c3087-fcce-4e3c-98df-e4bd1534dbeb\Work\System.Runtime.Tests\Unzip\native\xunit.console.netcore.dll", Status: 0xc0000135
6960:5834 @ 206029390 - LdrpInitializeProcess - ERROR: Walking the import tables of the executable and its static imports failed with status 0xc0000135
6960:5834 @ 206029390 - _LdrpInitialize - ERROR: Process initialization failed with status 0xc0000135
6960:5834 @ 206029390 - LdrpInitializationFailure - ERROR: Process initialization failed with status 0xc0000135
```
... which then exits before nearly anything happens (i.e. can't even run 'g' in the debugger).

**Comment by danmosemsft** [comment]
@MattGal  I would love to know how to get on such machines to save you doing this in future.

**Comment by MattGal** [comment]
Since this problem isn't ARM specific, you can grab repro bits for it locally using the core app from the share I've sent offline to you.  (I did all my debugging on an AMD64 machine; the libc problem seems global)

**Comment by danmosemsft** [comment]
@tommcdon do you still plan to take a look at this? Right now we are running in the dark - no test data for AOT for some time.

**Comment by pjanotti** [comment]
@luqunl q: on my machine I need to change tests.targets to use `-usedefaultpinvoke:false`, otherwise, I hit the error `libc.dll not found`. Why the same is not needed for CI?

**Comment by pjanotti** [comment]
I'm also having to manually delete some files when I try to run it multiple times.

**Comment by luqunl** [comment]
@pjanotti , Do you know how to get latest CI result for UAPAOT?

**Comment by pjanotti** [comment]
I just go to the front page of mc and click on the UWP ILC for CoreFx, right now it gets me to https://mc.dot.net/#/product/netcore/21/source/prodcon~2Fcorefx~2Frelease~2F2.1~2F/type/test~2Ffunctional~2Filc~2F/build/20180801.01/workItem/System.ComponentModel.Composition.Tests.Execution/wilog

**Comment by MattGal** [comment]
Just a nit to add there;  If you mean CI as in "Tests run on a per-checkin, or pre-checkin basis" we don't have this currently, because daily builds of the toolchain aren't generally available for folks outside the Microsoft organization.   I'd generally call those official builds.

**Comment by pjanotti** [comment]
Thanks for the nit @MattGal I didn't mean the ones triggered by PR.

**Comment by ViktorHofer** [comment]
@MattGal though the tests are failing I see tests run in the recent uapaot infrastructure logs: https://mc.dot.net/#/product/netcore/21/source/prodcon~2Fcorefx~2Frelease~2F2.1~2F/type/test~2Ffunctional~2Filc~2F/build/20180810.01/workItem/System.Text.RegularExpressions.Tests.Execution/wilogs

Is this already addressed?

**Comment by MattGal** [comment]
Yes, since at least compilation finished and it ran XUnit the libc.dll issue at least seems fixed.  Still failing, of course.

**Comment by ViktorHofer** [comment]
> Yes, since at least compilation finished and it ran XUnit the libc.dll issue at least seems fixed

Hmm, not sure how the libc.dll issue can be fixed if I still hit it locally with uapaot on x64.

**Comment by MattGal** [comment]
I don't know; I'm just basing that off the log you sent.  XUnit clearly ran here.

**Comment by A-And** [comment]
@ViktorHofer - I'm still hitting the `libc.dll` issue. Has this been addressed or have you managed to find a workaround?

**Comment by ViktorHofer** [comment]
Sorry for the delay, I had to first fix another related issue to be sure that this isn't caused by some recent xunit changes. I just tried it again and I'm also still hitting the libc.dll and VCRUNTIME140_APP.dll issue.

ping @luqunl @cshung

**Comment by luqunl** [comment]
Please CC: @jeffschwMSFT , @AaronRobinsonMSFT

**Comment by ViktorHofer** [comment]
Following up offline.

**Comment by ViktorHofer** [comment]
To give an update here, the current runner (xunit.console) doesn't work well with UAP and ILC. There are multiple options here, the ideal one would be to just use vstest.console.exe with ILC enabled which also runs in an AppContainer but unfortunately their UWP runner is outdated (https://github.com/Microsoft/vstest/issues/1884).
Our best option to unblock this asap is to use the fork of the official runner (https://github.com/dotnet/arcade/pull/1627) and use conditional compilation to remove parts which don't work well with UAP and ILC.

**Comment by ViktorHofer** [comment]
Finally this is now fixed in master and I will backport it into the UWP 6.2 branch.
